ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlServerEndpoints.java

```diff
@@ -296,20 +300,39 @@ public void executeWebsocketStream(final ServerWebSocket webSocket, final MultiM
       final WorkerExecutor workerExecutor,
       final ApiSecurityContext apiSecurityContext,
       final Context context) {
-
     executeOnWorker(() -> {
       final KsqlSecurityContext ksqlSecurityContext = ksqlSecurityContextProvider
           .provide(apiSecurityContext);
       try {
-        wsQueryEndpoint
-            .executeStreamQuery(webSocket, requestParams, ksqlSecurityContext, context);
+        wsQueryEndpoint.executeStreamQuery(
+            webSocket,
+            requestParams,
+            ksqlSecurityContext,
+            context,
+            getTokenTimeout(apiSecurityContext.getPrincipal())
+        );
       } finally {
         ksqlSecurityContext.getServiceContext().close();
       }
       return null;
     }, workerExecutor);
   }
 
+  private Optional<Long> getTokenTimeout(final Optional<KsqlPrincipal> principal) {
```

Comment:
I'm afraid you'll need this implementation as part of the `confluent-security-plugins` to access the `JwtPrincipal`. The security plugins is not open-source, so we cannot ship it with it.

One good thing is this timeout will only available on RBAC environments, so we can reuse the `KsqlSecurityExtension` to access security features. There's one impl. in  the `confluent-security-plugins`. The `KsqlSecurityExtension` is part of ksqldb as interface and dummy impls. only.

Reply:
I added a `KsqlAuthTokenProvider` to the extensions. The security-plugins changes are here: https://github.com/confluentinc/confluent-security-plugins/pull/483